.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,47 @@
+name: golangci-lint
```

Comment:
I suggest changing the `name` to `Lint` and add one more job to scan vulnerability:

```yaml
  # TODO: Should be removed after golangci-lint integrated with vuln
  vulnerability:
    name: Vulnerabilities Checks
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3

      - uses: actions/setup-go@v3
        with:
          go-version: "^1.20.0"
          cache: true

      - uses: actions/cache@v3
        with:
          path: ~/go/pkg/mod
          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
          restore-keys: |
            ${{ runner.os }}-go-

      - name: Scan vulnerability
        run: |
          go install golang.org/x/vuln/cmd/govulncheck@latest
          govulncheck ./...
```

More about vuln: https://go.dev/blog/vuln

And `golangci-lint` is working on to add the integration support with [`vuln`](https://github.com/golangci/golangci-lint/issues/3094), you can track the progress [here](https://github.com/golangci/golangci-lint/pull/3199)